### ISSUE TITLE
parallel builds not quite there yet

### ISSUE THREAD

**Issue by edsantiago** [containers/podman#13626]
Looks like #13404 still needs some work. It's flaking. Example 1 was a CI flake in #13536, I can't find a link now. Example 2 happened just a few minutes ago in [f35 rootless netavark](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/6099794304172032/html/sys-podman-fedora-35-rootless-host-netavark.log.html#t--00137).

**Comment by flouthoc** [comment]
@edsantiago I tried a few things here https://github.com/containers/podman/pull/13536#issuecomment-1072967446 but I was not able to replicate the build flakes but I can see that untagging is stll using `SetNames` however it should use new `RemoveNames` which might cause these flakes, maybe we need to remove `rmi` from the test and use built image count in the test.

**Comment by flouthoc** [comment]
@edsantiago One of my suspect is also `wait` call in the test, does `wait` guarantee that a build process was completed, however i'm not able to repro this on my local so this is just my suspect.

But I'll play again with this tests on my machine in really high quantity and chaos to see if i could reproduce any flakes.

Just for reference actual verifier and reproducer which customer reported on bz is this, it would be good if we could get test close to this.
```bash
#!/bin/bash
x=1
sudo ./podman rmi -af
rm -f log*.*
sudo ./podman build -t first . &> logfirst.log
while [ $x -le 30 ]
do
  echo "$x times"
  sudo ./podman build --log-level debug -t $x . &> log$x.log &
  x=$(( $x + 1 ))
#  sleep 1
done
```

**Comment by flouthoc** [comment]
Yup I think its the race in untagging which happens while calling `rmi` but not a build race a separate issue i think but worth fixing, I'll rectify test so they are only `build` focused and don't use other operations.

**Comment by edsantiago** [comment]
If the race is in `rmi`, an argument could be made that the crucial task is to fix the race, not patch the test. (i.e., this may be a really useful test to *keep*)

**Comment by flouthoc** [comment]
@edsantiago Yes that must fixed across all code base. Entire code should stop using function `SetNames` but to make sure we don't regress parallel build test i was thinking of something like this in the test for now but I to verify the below diff first.

```diff
--- a/test/system/070-build.bats
+++ b/test/system/070-build.bats
@@ -231,7 +231,9 @@ EOF
 
     # Now delete all built images. If any image wasn't built, rmi will fail
     # and test will fail.
-    run_podman rmi $(seq --format 'i%02g' 1 $count)
+    for i in $(seq --format 'i%02g' 1 $count); do
+       run_podman rmi $i
+    done
 }
```

**Comment by flouthoc** [comment]
I think the flakes identified last time were purely out of OOM in CI env, @edsantiago Should we close this ?

**Comment by edsantiago** [comment]
@flouthoc I seem to have missed something. Could you point me at discussion or analysis suggesting your OOM idea?

**Comment by flouthoc** [comment]
> @flouthoc I seem to have missed something. Could you point me at discussion or analysis suggesting your OOM idea?

@edsantiago Are we still facing this anywhere ? I think we captured logs on rhel gating it was failing because of OOM with error like `fatal error: out of memory allocating heap`. I think similar is happening here but we will have to capture logs to confirm as of now we redirect log to `/dev/null`.

**Comment by edsantiago** [comment]
Test last flaked on 05-23:

### [Netavark] 136 podman parallel build should not race
* fedora-35 : Netavark sys podman root
  * PR #14326
    * [05-23 15:51](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/4623092222787584/html/sys-podman-fedora-35-root-host-netavark.log.html#t--00136)

Looking at the [journal log](https://cirrus-ci.com/task/4623092222787584?logs=journal#L0), searching for "OOM" or "memory" or "heap" or "allocating" or "fatal" all find no hits. (I may be searching the wrong log, though). I'm reluctant to close this until the issue is better understood.

**Comment by flouthoc** [comment]
> searching for "OOM" or "memory" or "heap" or "allocating" or "fatal" all find no hits. (I may be searching the wrong log, though)

Yes I think in order to get correct log we need to remove piping it to `/dev/null` , I'll figure out a way to get logs from one of the failed builds. But I agree we should not close this till we are sure :)

**Comment by rhatdan** [comment]
@flouthoc any progress on this?

**Comment by flouthoc** [comment]
I think its not reported in any recent runs but i am not sure @edsantiago could confirm it better. I am not sure if this persists on upstream since original BZ author ackd and closed the BZ on the provided fix. If we see this again in any of the CI runs we can instrument the test to collect the logs and debug it further.

**Comment by edsantiago** [comment]
I see no flakes since the 05-23 one reported above, and I don't remember enough about this issue to make an informed suggestion.
